optimiser/optimisation_problem.py

```diff
@@ -0,0 +1,220 @@
+    # Define the problem
+    problem = pl.LpProblem("BatteryOptimisation", pl.LpMinimize)
+
+    # Time slices
+    T = range(1, 25)
+
+    TIMESLICE_SIZE = 0.5
+
+    # Renewable Generation Constants
+    MAX_RENEWABLE_GENERATION = 10  # kWh
+
+    # Battery Constants
+    MAX_BATTERY_DISCHARGE_RATE = 5
+    MAX_BATTERY_CHARGE_RATE = 5
+    MAX_BATTERY_CAP = 100
+
+    # Costs Constants
+    UNIT_PRICE_GRID_ELECTRICITY = 0.3
+    UNIT_PRICE_BATTERY_ELECTRICITY = 0
+    UNIT_PRICE_RENEWABLE_ELECTRICITY = 0
+    UNIT_PRICE_BATTERY_SALE_PRICE = 0.1
+
+    demand = {}
+    for t in T:
+        demand[t] = 100
+
+    # ---- Define variables ----
+    # Renewable electricity flow
+    renewable_electricity_to_house = LpVariable.dicts(
+        name="renewable_electricity_to_house_{t}",
+        indexs=T,
+        lowBound=0,
+        cat="Continuous",
+    )
+    renewable_electricity_to_battery = LpVariable.dicts(
+        name="renewable_electricity_to_battery_{t}",
+        indexs=T,
+        lowBound=0,
+        cat="Continuous",
+    )
+    total_renewable_generation = LpVariable.dicts(
+        name="total_renewable_generation_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+
+    # Battery electricity flow
+    battery_electricity_to_house = LpVariable.dicts(
+        name="battery_electricity_to_house_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    battery_electricity_to_grid = LpVariable.dicts(
+        name="battery_electricity_to_grid_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    electricity_to_battery = LpVariable.dicts(
+        name="electricity_to_battery_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+
+    # Grid electricity flow
+    grid_electricity_to_house = LpVariable.dicts(
+        name="grid_electricity_to_house_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    grid_electricity_to_battery = LpVariable.dicts(
+        name="grid_electricity_to_battery_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+
+    # Fuel Costs
+    fuel_costs = LpVariable.dicts(
+        name="fuel_costs_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    battery_electricity_costs = LpVariable.dicts(
+        name="battery_electricity_costs_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    grid_electricity_costs = LpVariable.dicts(
+        name="grid_electricity_costs_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    renewable_eletricity_costs = LpVariable.dicts(
+        name="renewable_eletricity_costs_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    battery_to_grid_sales = LpVariable.dicts(
+        name="battery_to_grid_sales_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    # Battery state of charge
+    battery_state_of_charge = LpVariable.dicts(
+        name="battery_state_of_charge_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
+    battery_degradation = LpVariable.dicts(
+        name="battery_degradation_{t}", indexs=T, lowBound=0, cat="Continuous"
+    )
```

Comment:
Would be good to define model variables in their own module as well - also raise in another issue ticket

Reply:
https://github.com/davidwickh/Home-Battery-Optimiser/issues/5